### ISSUE TITLE
Change in simulate functionality

### ISSUE THREAD

**Issue by bdonon** [MarvinLer/pypownet#13]
For the presentation of August 30, we should have a function simulate that would be able to try an action, get the full reward, and then go back to the state before the action.
What's currently being done is computing a partial reward with a new topology, but with the same injections.

**Comment by MarvinLer** [comment]
Added in 8f456ac.
The class Game now has a simulate method which performs given an action:
* store the before-action topology and id of scenario (i.e. set of injections)
* apply the action + load the new injections and then compute loadflow + cascading failure: if this fails, then raise an error, otherwise returns the destination state
* put the initial topology back on + loads the initial set of injections
* compute loadflow + cascading failure which sould lead to the same loadflow as before the simulation

The method returns the simulated final observation if succeeded. The RunEnv class has also a simulate method that handle the rewards computations based on the response of the above method.
This needs some testing

**Comment by bdonon** [comment]
I got a topological action that resulted in a very low reward, even though the simulate function predicted a "reasonable" value:

expected reward : -0.9211436526345437
Action chosen: changing topology of node 9
[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0.]
Simulation step 30:
 ->            reward: -43.92050497422201

**Comment by bdonon** [comment]
The action taken corresponds to change the node of one element from substation 9. Which results in some element being disconnected.
And the behavior of simulate and step is not the same in this case

**Comment by MarvinLer** [comment]
Fixed in 9549d6a.
Thanks for the bug catch-up! The bug was that the mapping function, which allows to pack the values of the elements of a same substation in the topological vector was missing some part, which provoked node-splitting instead of line status switches. Thus, the distance to the reference grid subsreward was not 0, hence the difference in the expected/real results.
I also added a TreeSearchLineServiceStatus agent (could be generalized to any depth) for testing purposes.

I tested on the case14, the results are as identical to expectations. Also, it takes ~4s per timestep with this exhaustive one depth method.
